orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -102,5 +102,112 @@ Status DropoutGrad<T1, T2>::ComputeInternal(OpKernelContext* context) const {
 
   return Status::OK();
 }
+
+#define REGISTER_BIAS_DROPOUT_KERNEL_TYPED(T1, T2)                   \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                     \
+      BiasDropout,                                                   \
+      kMSDomain,                                                     \
+      1,                                                             \
+      T1##_##T2,                                                     \
+      kCudaExecutionProvider,                                        \
+      KernelDefBuilder()                                             \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T1>())    \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<T2>())   \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()) \
+          .InputMemoryType<OrtMemTypeCPUInput>(3)                    \
+          .InputMemoryType<OrtMemTypeCPUInput>(4),                   \
+      BiasDropout<T1, T2>);
+
+REGISTER_BIAS_DROPOUT_KERNEL_TYPED(MLFloat16, MLFloat16)
```

Comment:
>REGISTER_BIAS_DROPOUT_KERNEL_TYPED [](start = 0, length = 34)

can probably make the BiasDropout class not templated and use the MLTypeCallDispatcher helper
[https://github.com/microsoft/onnxruntime/blob/12367a6b11e96e70112026cb7975f7035c7ad0ed/include/onnxruntime/core/framework/data_types_internal.h#L300](https://github.com/microsoft/onnxruntime/blob/12367a6b11e96e70112026cb7975f7035c7ad0ed/include/onnxruntime/core/framework/data_types_internal.h#L300)